ml-agents/mlagents/trainers/policy.py

```diff
@@ -79,14 +80,33 @@ def _load_graph(self):
                             .format(self.model_path))
             self.saver.restore(self.sess, ckpt.model_checkpoint_path)
 
-    def evaluate(self, brain_info):
+    def evaluate(self, brain_info: BrainInfo):
         """
         Evaluates policy for the agent experiences provided.
         :param brain_info: BrainInfo input to network.
         :return: Output from policy based on self.inference_dict.
         """
         raise UnityPolicyException("The evaluate function was not implemented.")
 
+    def take_action(self, brain_info: BrainInfo) -> ActionInfo:
```

Comment:
Coming from the perspective of the `Policy` class as implementing a `policy` mathematical function, I think it makes sense to `take_action` here. It reminds me of say a key-value store where you feed in observations and get back actions.  Going from that perspective, what are your thoughts on renaming `take_action` to something like `get_action`? This way it is clear that this is very distinct in behavior from say `take_step` which exists in the same code.

Reply:
@eshvk I think that's a very good suggestion.  No action is being taken here, we are just looking up the correct action info for a given set of obs/reward :)

Reply:
Yeah, exactly! Please feel free to incorporate it now in this PR or we can table it for a future PR.